science/gcal/files/patch-trap-remove-may-month-check-fix.diff

```diff
@@ -33,36 +34,9 @@
 -
 -      return (mayname);
 -    }
-+// MHB 2020-11-27 Causes trap when running Big Sur. Not sure required.
-+//  if (month == 5)
-+//    {
-+//      static char *mayname;
-+//      static Bool is_initialized = FALSE;
-+//
-+//
-+//      /*
-+//         Now let's remove the trailing BLANK character
-+//         from the complete text for the month MAY.
-+//       */
-+//      if (!is_initialized)
-+//	{
-+//	  register int len;
-+//	  auto char *s = _(name[month]);
-+//
-+//
-+//	  len = (int) strlen (s) - 1;
-+//	  mayname = (char *) my_malloc (len,
-+//					ERR_NO_MEMORY_AVAILABLE,
-+//					__FILE__, ((long) __LINE__) - 2L,
-+//					"mayname", 0);
-+//	  strcpy (mayname, s);
-+//	  while (*(mayname + len) == ' ')
-+//	    *(mayname + len--) = '\0';
-+//	  is_initialized = TRUE;
-+//	}
-+//
-+//      return (mayname);
-+//    }
++  if (month == 5) {
++      return("May");
```

Comment:
You can't just return string "May" here because month names can be localized (translated to another language), that's why we need all these N_(). Also, that extra space (which is normally deleted here in code removed by you) is needed to distinguish abbreviated month name from full month name, which may be the same for May in English, but not in other languages.

Reply:
Hi Oleg. Yes you are right. I did not take into account localization. I have added what I think identifies it as text to be localized. 

The loop seems so complicated for what should be something really simple. And it was quite the job trying to work out that it was a cause of the segfault.

I am new to this process of getting code applied upstream and then making its way down through the macports system. Did you email Giuseppe your patch? And then we wait for him to do a release. And then we have to redo the port file and patch (if necessary). And then we have to update the pull request. And then ask j.hafey to accept the patch.